### PR TITLE
Fix LDAP service account creation

### PR DESCRIPTION
**Pull request by donatello**
## Description
- when user has only group permissions

- fixes regression from ac74237f0 (#13657)

- Fixes https://github.com/minio/console/issues/1291

## Motivation and Context

Bug fix.

## How to test this PR?

See linked issue

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression (#13657 )
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

File: cmd/sts-handlers_test.go [hunk 1]
```diff
@@ -443,6 +444,106 @@ func (s *TestSuiteIAM) TestLDAPSTSServiceAccounts(c *check) {
 	c.assertSvcAccDeletion(ctx, s, userAdmClient, value.AccessKeyID, bucket)
 }
 
+// In this test, the parent users gets their permissions from a group, rather
+// than having a policy set directly on them.
+func (s *TestSuiteIAM) TestLDAPSTSServiceAccounts2(c *check) {
```
Comment (harshavardhana) [inline cmd/sts-handlers_test.go]:
```suggestion
func (s *TestSuiteIAM) TestLDAPSTSServiceAccountsWithGroups(c *check) {
```

File: cmd/sts-handlers_test.go [hunk 2]
```diff
@@ -225,6 +225,7 @@ func TestIAMWithLDAPServerSuite(t *testing.T) {
 				suite.SetUpLDAP(c, ldapServer)
 				suite.TestLDAPSTS(c)
 				suite.TestLDAPSTSServiceAccounts(c)
+				suite.TestLDAPSTSServiceAccounts2(c)
```
Comment (harshavardhana) [inline cmd/sts-handlers_test.go]:
```suggestion
				suite.TestLDAPSTSServiceAccountsWithGroups(c)
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by krisis — APPROVED** [review]

**Pull request merged**